history/src/postgres/postgres_block.rs

```
@@ -1,45 +1,85 @@
             .unwrap_or(None);
 
         Self {
-            leader_id: value.leader_id,
-            blockhash: value.blockhash,
+            blockhash: value.blockhash.clone(),
             block_height: value.block_height as i64,
             slot: value.slot as i64,
             parent_slot: value.parent_slot as i64,
             block_time: value.block_time as i64,
-            commitment_config: commitment as i8,
-            previous_blockhash: value.previous_blockhash,
+            previous_blockhash: value.previous_blockhash.clone(),
             rewards,
         }
     }
 }
+
+impl PostgresBlock {
+    pub fn create_statement(schema: &String) -> String {
+        format!(
+            "
+            CREATE TABLE {}.BLOCKS (
+                slot BIGINT PRIMARY KEY,
+                blockhash STRING NOT NULL,
+                leader_id STRING,
+                block_height BIGINT NOT NULL,
+                parent_slot BIGINT NOT NULL,
+                block_time BIGINT NOT NULL,
+                previous_blockhash STRING NOT NULL,
+                rewards STRING,
+            );
+        ",
+            schema
+        )
+    }
+
+    pub async fn save(&self, postgres_session: &PostgresSession, schema: &String) {
+        let mut query = format!(
+            r#"
+            INSERT INTO {}.BLOCKS (slot, blockhash, block_height, parent_slot, block_time, previous_blockhash, rewards) VALUES 
+        "#,
+            schema
+        );
+
+        let mut args: Vec<&(dyn ToSql + Sync)> = Vec::with_capacity(NB_ARUMENTS);
+        args.push(&self.slot);
+        args.push(&self.blockhash);
+        args.push(&self.block_height);
+        args.push(&self.parent_slot);
+        args.push(&self.block_time);
+        args.push(&self.previous_blockhash);
+        args.push(&self.rewards);
+
+        PostgresSession::multiline_query(&mut query, NB_ARUMENTS, 1, &[]);
+        if let Err(e) = postgres_session.execute(&query, &args).await {
+            log::error!(
```

The same perhaps it's better not to hide the error and return it.